tests/console/sshd.pm

```diff
@@ -80,13 +80,32 @@ sub run {
     select_console 'root-console';
 
     # Generate RSA key for SSH and copy it to our new user's profile
-    assert_script_run "ssh-keygen -t rsa -P '' -C 'localhost' -f ~/.ssh/id_rsa";
-    assert_script_run "install -m 1700 -o $ssh_testman -d /home/$ssh_testman/.ssh";
+    assert_script_run "ssh-keygen -t rsa -P '' -C 'root\@localhost' -f ~/.ssh/id_rsa";
+    assert_script_run "sudo -u $ssh_testman ssh-keygen -t rsa -P '' -C '$ssh_testman\@localhost' -f /home/$ssh_testman/.ssh/id_rsa";
     assert_script_run "install -m 0644 -o $ssh_testman ~/.ssh/id_rsa.pub /home/$ssh_testman/.ssh/authorized_keys";
+    assert_script_run "cat /home/$ssh_testman/.ssh/id_rsa.pub >> /home/$ssh_testman/.ssh/authorized_keys";
 
     # Test non-interactive SSH and after that remove RSA keys
     assert_script_run "ssh -4v $ssh_testman\@localhost bash -c 'whoami | grep $ssh_testman'";
 
+    # Port forwarding (bsc#1131709 bsc#1133386)
+    assert_script_run "( ssh -vv -L 4242:localhost:22 $ssh_testman\@localhost sleep 9999 & )";
+    assert_script_run "( ssh -vv -R 0.0.0.0:5252:localhost:22 $ssh_testman\@localhost sleep 9999 & )";
+    assert_script_run "ssh-keyscan -p 4242 localhost >> ~/.ssh/known_hosts";
+    assert_script_run "ssh-keyscan -p 5252 localhost >> ~/.ssh/known_hosts";
+
+    assert_script_run "ssh -p 4242 $ssh_testman\@localhost whoami";
+    assert_script_run "ssh -p 5252 $ssh_testman\@localhost whoami";
+    assert_script_run "ssh -p 4242 $ssh_testman\@localhost 'ssh-keyscan -p 22 localhost 127.0.0.1 ::1 >> ~/.ssh/known_hosts'";
+    assert_script_run "ssh -p 4242 $ssh_testman\@localhost 'ssh-keyscan -p 4242 localhost 127.0.0.1 ::1 >> ~/.ssh/known_hosts'";
+    assert_script_run "ssh -p 4242 $ssh_testman\@localhost 'ssh-keyscan -p 5252 localhost 127.0.0.1 ::1 >> ~/.ssh/known_hosts'";
+
+    assert_script_run "ssh -p 4242 -tt $ssh_testman\@localhost ssh -tt $ssh_testman\@localhost whoami";
+    assert_script_run "ssh -t -o ProxyCommand='ssh $ssh_testman\@localhost nc localhost 4242' $ssh_testman\@localhost whoami";
+    if (is_opensuse || is_sle('15+')) {
+        assert_script_run "ssh -J $ssh_testman\@localhost:4242 $ssh_testman\@localhost whoami";
```

Comment:
This is broken on openSUSE Leap 42.3... `unknown option -J`
https://openqa.opensuse.org/tests/929974#step/sshd/80